spec/src/main/asciidoc/core/invokers.asciidoc

```diff
@@ -0,0 +1,175 @@
+If the target method returns normally, its return value is returned, unless the target method is declared `void`, in which case `null` is returned.
+If the target method throws an exception, it is wrapped into `InvocationTargetException` and that is thrown.
+
+If the target method is not `static` and `instance` is `null`, a `RuntimeException` is thrown.
+If the target method is not `static` and the `instance` is not <<invoker_assignability,assignable>> to the target bean, a `RuntimeException` is thrown.
+
+Correspondence between given `arguments` and declared parameters of the target method is positional: the Nth element of the `arguments` array is passed as the Nth argument to the target method.
+If the target method is a variable arity method, the last element of the `arguments` array corresponds to the variable arity parameter (and therefore must be an array).
+
+If the target method declares no parameter, `arguments` are ignored.
+If the target method declares any parameter and `arguments` is `null`, `RuntimeException` is thrown.
+If the `arguments` array has fewer elements than the number of parameters of the target method, `RuntimeException` is thrown.
+If the `arguments` array has more elements than the number of parameters of the target method, the excess elements are ignored.
+If some of the `arguments` is not <<invoker_assignability,assignable>> to the corresponding parameter of the target method, `RuntimeException` is thrown.
+
+If the declared type of a parameter of the target method is a primitive type, an unboxing conversion is performed on the corresponding argument, a widening primitive conversion is performed on the result if needed, and the result is passed.
+If the declared type of a parameter of the target method is not a primitive type, the corresponding argument is passed as is.
+
+// TODO when the `InvokerBuilder` applies transformations, some of the requirements above are no longer strictly necessary, we should reflect that in this text somehow
+
+==== Example
+
+To illustrate how method invokers work, let's take a look at an example.
+Say that the following bean exists in an application and has a method that you, the framework author, want to invoke indirectly:
+
+[source,java]
+----
+@Dependent
+public class MyService {
+    public String hello(String name) {
+        return "Hello " + name + "!";
+    }
+}
+----
+
+In a CDI extension, you obtain an `InvokerBuilder` for the `hello()` method and use it to build an invoker.
+In a portable extension, this results in an invoker which should be stored for later usage:
+
+[source,java]
+----
+InvokerBuilder<Invoker<MyService, String>> builder = ...;
+Invoker<MyService, String> invoker = builder.build();
+----
+
+In a build compatible extension, this results in an opaque token that materializes as an `Invoker` at application runtime:
+
+[source,java]
+----
+InvokerBuilder<InvokerInfo> builder = ...;
```

Comment:
I think it needs more explanation with the ...

Reply:
The build compatible extensions case is covered below. The case for portable extensions is covered in the CDI Full section. It would be great to avoid ellipses, but that would require showing much more context, which is detrimental to the purpose of this particular example.

Reply:
I suggest adding a comment to link to the relevant section in the doc.

Reply:
Good point, will do!

Reply:
Done.